lib/validate/net.go

```diff
@@ -46,30 +48,55 @@ func KubernetesSubnetsFromStrings(podCIDR, serviceCIDR string) error {
 		}
 	}
 
+	// make sure podSubnetSize is valid
+	if podSubnetSize != "" {
+		subnetSize, err = strconv.Atoi(podSubnetSize)
+		if err != nil || subnetSize < 1 || subnetSize > 32 {
+			return trace.BadParameter("invalid pod subnet size: %q", podSubnetSize)
+		}
+
+		// The minimum subnet size accepted by flannel is /28:
+		// https://github.com/gravitational/flannel/blob/master/subnet/config.go#L70-L74
```

Comment:
nit: should use a specific commit as the master label is a moving target and the comment might get invalid soon